DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -503,7 +504,43 @@ internal DiscordGuild()
         }
 
         #region Guild Methods
+        
+        /// <summary>
+        /// Gets guild's icon URL, in requested format and size.
+        /// </summary>
+        /// <param name="imageFormat">The image format of the icon to get.</param>
+        /// <param name="imageSize">The maximum size of the icon. Must be a power of two, minimum 16, maximum 4096.</param>
+        /// <returns>The URL of the guild's icon.</returns>
+        public string GetIconUrl(ImageFormat imageFormat, ushort imageSize = 1024)
+        {
+
+            if (imageFormat == ImageFormat.Unknown)
+                throw new ArgumentException("You must specify valid image format.", nameof(imageFormat));
+
+            // Makes sure the image size is in between Discord's allowed range.
+            if (imageSize < 16 || imageSize > 4096)
+                throw new ArgumentOutOfRangeException("Image Size is not in between 16 and 4096: " + nameof(imageSize));
```

Comment:
I'm not sure about coding standards in D#+, but won't it be better to utilize [this constructor](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=net-6.0#system-argumentoutofrangeexception-ctor(system-string-system-object-system-string)) which will set name of param, and its actual value.
```suggestion
            if (imageSize < 16 || imageSize > 4096)
                throw new ArgumentOutOfRangeException(nameof(imageSize), imageSize, "Image Size is not in between 16 and 4096.");
```